### ISSUE TITLE
unsupervised graph-based loss function

### ISSUE THREAD

**Issue by bsaghafi** [rusty1s/pytorch_geometric#1507]
Is there any unsupervised graph-based loss function available in the package? I am trying to use GraphSAGE in an unsupervised setting, therefore need to use a loss function only based on neighborhood similarity, like the one defined in equation (1) in the [GraphSAGE paper](https://arxiv.org/pdf/1706.02216.pdf).

**Comment by rusty1s** [comment]
Hi, please see https://github.com/rusty1s/pytorch_geometric/issues/1420.
